Bug: `-fae` flag causes incorrect path resolution in plugin goals

### Affected version

1.0.6

### Bug description

## Summary
When using `mvnd -fae` (intended as shorthand for `--fail-at-end`), the flag is incorrectly parsed as separate flags (`-f`, `-a`, `-e`), causing plugin goals to receive corrupted or incorrect working directory/base path values. This does not occur with Maven's `mvn -fae` or when using `mvnd --fail-at-end`.

## Expected Behavior
`mvnd -fae` should be equivalent to `mvnd --fail-at-end`, and plugins should execute with the correct base paths regardless of which form is used.

## Actual Behavior
When running `mvnd -fae`, plugins receive incorrect directory paths. For example, the `xml-maven-plugin:transform` goal fails with:

```
[ERROR] The directory /path/to/project/ae/docs, 
which is a base directory of a ValidationSet or TransformationSet, does not exist.
```

The spurious `ae/` directory component appears to come from the misinterpreted `-a` and `-e` flags.

## Steps to Reproduce
1. Use any Maven project with a plugin that operates on file paths (e.g., `xml-maven-plugin`)
2. Run: `mvnd -fae clean install`
3. Observe path corruption in plugin execution

## Workaround
Use the long form flag instead:
```bash
mvnd --fail-at-end clean install
```

## Environment
- mvnd: 1.0.6 darwin-aarch64
- Maven: 3.9.16
- Java: 25.0.3 (Temurin)
- OS: macOS 26.5.1, aarch64 (but also experienced on Linux CI runners)

## Additional Context
- `mvn -fae` works correctly with standard Maven
- `mvnd --fail-at-end` works correctly
- This suggests a flag-parsing regression specific to mvnd's handling of compound short flags

The issue is not immediately obvious to users switching from Maven to mvnd, making it a usability/documentation concern at minimum.

--- a/client/src/test/java/org/mvndaemon/mvnd/client/EnvironmentTest.java
+++ b/client/src/test/java/org/mvndaemon/mvnd/client/EnvironmentTest.java
@@ -53,6 +53,26 @@ public class EnvironmentTest {
         assertEquals("", Environment.MAVEN_DEFINE.removeCommandLineOption(list("--define")));
     }
 
+    @Test
+    void mavenFileArguments() {
+        assertEquals("pom.xml", Environment.MAVEN_FILE.removeCommandLineOption(list("-f", "pom.xml")));
+        assertEquals("pom.xml", Environment.MAVEN_FILE.removeCommandLineOption(list("-fpom.xml")));
+        assertEquals("pom.xml", Environment.MAVEN_FILE.removeCommandLineOption(list("-f=pom.xml")));
+        assertEquals("pom.xml", Environment.MAVEN_FILE.removeCommandLineOption(list("--file", "pom.xml")));
+        assertEquals("pom.xml", Environment.MAVEN_FILE.removeCommandLineOption(list("--file=pom.xml")));
+    }
+
+    @Test
+    void mavenFailModeArgumentsAreNotMavenFileArguments() {
+        for (String failMode : Arrays.asList("-fae", "-ff", "-fn")) {
+            List<String> args = list(failMode, "verify");
+
+            Assertions.assertFalse(Environment.MAVEN_FILE.hasCommandLineOption(args));
+            Assertions.assertNull(Environment.MAVEN_FILE.removeCommandLineOption(args));
+            assertEquals(list(failMode, "verify"), args);
+        }
+    }
+
     private List<String> list(String... items) {
         return new ArrayList<>(Arrays.asList(items));
     }

--- a/common/src/main/java/org/mvndaemon/mvnd/common/Environment.java
+++ b/common/src/main/java/org/mvndaemon/mvnd/common/Environment.java
@@ -314,6 +314,9 @@ public enum Environment {
     MVND_CANCEL_CONNECT_TIMEOUT("mvnd.cancelConnectTimeout", null, "3 seconds", OptionType.DURATION, Flags.NONE),
     ;
 
+    // Maven defines these as complete short options, not as "-f" with an attached value.
+    private static final Set<String> MAVEN_FILE_SHORT_OPTION_CONFLICTS = Set.of("-fae", "-ff", "-fn");
+
     static Properties properties;
 
     public static void setProperties(Properties properties) {
@@ -477,7 +480,8 @@ public enum Environment {
 
     public boolean hasCommandLineOption(Collection<String> args) {
         final String[] prefixes = getPrefixes();
-        return args.stream().anyMatch(arg -> Stream.of(prefixes).anyMatch(arg::startsWith));
+        return args.stream()
+                .anyMatch(arg -> Stream.of(prefixes).anyMatch(prefix -> matchesCommandLineOption(arg, prefix)));
     }
 
     public String getCommandLineOption(Collection<String> args) {
@@ -493,7 +497,7 @@ public enum Environment {
         String value = null;
         for (Iterator<String> it = args.iterator(); it.hasNext(); ) {
             String arg = it.next();
-            if (Stream.of(prefixes).anyMatch(arg::startsWith)) {
+            if (Stream.of(prefixes).anyMatch(prefix -> matchesCommandLineOption(arg, prefix))) {
                 if (remove) {
                     it.remove();
                 }
@@ -501,7 +505,7 @@ public enum Environment {
                     value = "";
                 } else {
                     String opt = Stream.of(prefixes)
-                            .filter(arg::startsWith)
+                            .filter(prefix -> matchesCommandLineOption(arg, prefix))
                             .max(Comparator.comparing(String::length))
                             .get();
                     value = arg.substring(opt.length());
@@ -521,6 +525,13 @@ public enum Environment {
             }
         }
         return value;
+    }
+
+    private boolean matchesCommandLineOption(String arg, String prefix) {
+        if (this == MAVEN_FILE && MAVEN_FILE_SHORT_OPTION_CONFLICTS.contains(arg)) {
+            return false;
+        }
+        return arg.startsWith(prefix);
     }
 
     private String[] getPrefixes() {

